### PR TITLE
Add tracing button

### LINKED ISSUE #147: add button to start tracing a mask
The button should replace listening to the `t` char event.

Also once `Create Mask` button has been clicked the tracer widget should be deactivated

### PR DESCRIPTION
**Pull request by paskino**
Adds a button to enable/disable tracing on the `Mask` panel.

closes #147 

To be merged after #142

### CODE REVIEW

File: src/idvc/dvc_interface.py [hunk 1]
```diff
@@ -1868,6 +1863,16 @@ def CreateMaskPanel(self):
         formLayout.setWidget(widgetno,QFormLayout.FieldRole, mp_widgets['extendMaskCheck'])
         widgetno += 1
 
+        # Add start tracing button
+        mp_widgets['start_tracing'] = QPushButton(groupBox)
+        mp_widgets['start_tracing'].setText("Start Tracing")
+        mp_widgets['start_tracing'].clicked.connect(self.ToggleTracing)
+        mp_widgets['start_tracing'].setCheckable(True)
+        mp_widgets['start_tracing'].setChecked(False)
+        # mp_widgets['start_tracing'].setToolTip("Press 't' and draw a region on the viewer to create a mask.")
```
Comment (lauramurgatroyd) [inline src/idvc/dvc_interface.py:1872]: could we set a tooltip to be consistent with other buttons and gui elements, perhaps:
```suggestion
        mp_widgets['start_tracing'].setToolTip("Begin drawing a region on the viewer to create a mask.")
```

File: src/idvc/dvc_interface.py [hunk 2]
```diff
@@ -1914,7 +1942,13 @@ def MaskWorker(self, type):
...
         self.create_progress_window("Loading", "Loading Mask")
         self.mask_worker.signals.progress.connect(self.progress)
-       
+
+        # disable tracing on the viewer
+        v.imageTracer.Off()
+        # set the button to unchecked and text to start tracing
+        self.mask_parameters['start_tracing'].setChecked(False)
+        self.mask_parameters['start_tracing'].setText("Start Tracing")    
+
```
Comment (lauramurgatroyd) [inline src/idvc/dvc_interface.py:1951]: This code seems to try to turn tracing off after a mask has been made? But I find tracing is still on once my mask has been made 😞

Comment (lauramurgatroyd) [reply src/idvc/dvc_interface.py:1951]: Now that I have tested more thoroughly I see that the only case where it doesn't turn off tracing and reset the button is the case where I already have a mask, then I trace and create a new mask.

If I have first cleared the mask, or it is a fresh run of the app and there is no previous mask displayed, then it resets the button and turns off the tracing just fine.

Comment (lauramurgatroyd) [reply src/idvc/dvc_interface.py:1951]: It isn't a big issue, just something I noticed when testing...

**Review by lauramurgatroyd — CHANGES_REQUESTED** [review]
I like it but I think we should update the help text to tell them to use the button:
image

Also found a minor issue (see comments)

File: src/idvc/dvc_interface.py (hunk 2 above)
Comment (paskino) [reply src/idvc/dvc_interface.py:1951]: I'll check it!

Comment (paskino) [reply src/idvc/dvc_interface.py:1951]: I tried as you say
> I already have a mask, then I trace and create a new mask.

My app first let me do it, but no result from my tracing. At a second attempt it crashed.

Comment (lauramurgatroyd) [reply src/idvc/dvc_interface.py]: Weird, I had no problems with crashing! Just that the tracing was/wasn't enabled at the right time

Comment (paskino) [reply src/idvc/dvc_interface.py]: I implemented a workaround with _the wise owl advise not to be naughty_

**Review by lauramurgatroyd — APPROVED** [review]

**Pull request merged**